LogBuffer.md

```diff
@@ -0,0 +1,97 @@
+|--------------------|-----------------------|--------------------|--------------------------------------------------------------------------------------------------|
+| `batchSize`        | number                | 50                 | The maximum number of log entries to buffer before sending them to the server.                   |
+| `dispatchInterval` | number                | 10000              | The interval (in milliseconds) at which the buffer should be flushed and logs sent to the server.|
+| `maxRetries`       | number                | 3                  | The maximum number of retries to attempt when sending logs to the server fails.                  |
+| `retryDelay`       | number                | 2000               | The base delay (in milliseconds) between retries when sending logs fails.                        |
+| `logLevel`         | LogEntryType or null  | LogEntryType.info  | The minimum log level to buffer and send to the server. Logs with a lower level will be ignored.  |
+| `enableCloudLogging` | boolean             | false              | Flag to enable or disable sending logs to the Acrolinx server.                                   |
+
+### Log Entry Types
+
+The LogBuffer class supports the following log entry types:
+
+- `LogEntryType.info`: Informational log entries.
+- `LogEntryType.warning`: Warning log entries.
+- `LogEntryType.error`: Error log entries.
+
+### Buffering and Sending Logs
+
+The `LogBuffer` class buffers the log entries until one of the following conditions is met:
+- The number of buffered log entries reaches the configured `batchSize`.
+- An `error` log entry is added to the buffer.
+- The configured `dispatchInterval` has elapsed.
+
+When any of these conditions is met, the LogBuffer will flush the buffer and send the logs to the Acrolinx server.
+
+
+### Retry Mechanism
+
+If sending logs to the server fails, the LogBuffer class will retry the operation up to the configured maxRetries. The retry delay is adaptive and increases exponentially with each retry attempt, up to a maximum of maxRetries * retryDelay.
+
+If the maximum number of retries is reached and the logs still cannot be sent, the logs are discarded, and an error message is logged to the console.
+
+### Local Console Logging
+
+In addition to sending logs to the server, the LogBuffer class also logs the messages to the local console based on their log level:
+
+- info logs are logged using `console.log`.
+- warning logs are logged using `console.warn`.
+- error logs are logged using `console.error`.
+
+This allows for local debugging and monitoring of the logged messages.
+
+### Example
+
+Here's an example of how to use the LogBuffer class in an application:
+
+```typescript
+import { LogBuffer, LogEntryType } from '@acrolinx/sdk-js';
+
+const logBuffer = new LogBuffer('https://example.com/api/logs', {
```

Comment:
Why logbuffer is not using acrolinx endpoint? It's our util to send info to our platform?